Image Swap (#1295)
spectrocloud/librarium: content/docs/04-clusters/06-cluster-management/50-image-swap.md

---
title: "Image Swap"
metaTitle: "Image Swap"
metaDescription: "Learn how to swap out images and registries through the image swap webhook exposed by Palette."
hideToC: false
fullWidth: false
---

import Tabs from 'shared/components/ui/Tabs';
import WarningBox from 'shared/components/WarningBox';
import InfoBox from 'shared/components/InfoBox';

# Overview

> Palette supports swapping out images and registries at the Kubernetes layer. Palette uses the *ImageSwap* webhook that is exposed by the [ImageSwap Mutating Admission Controller for Kubernetes](https://github.com/phenixblue/imageswap-webhook/blob/master/README.md). You can use this feature to override a specific number of container image registries or particular images. The following are a few common use cases the image swap feature is a good fit: <br /> <br />

Reviewer: ```suggestion
Palette supports swapping out images and registries at the Kubernetes layer. Palette uses the *ImageSwap* webhook that is exposed by the [ImageSwap Mutating Admission Controller for Kubernetes](https://github.com/phenixblue/imageswap-webhook/blob/master/README.md). You can use this feature to override a specific number of container image registries or particular images. The following are some common use cases for image swapping: <br /> <br />
```

- Avoid rate limit issues encountered with public images by pointing to an alternate image registry that caches public images. This is more common in an Enterprise setting.

- Changing the URL of a container registry, whether internal or external.

- Support air-gapped environments by redirecting public image requests to an internal registry.
 

 To use the image swap, specify an image swap configuration in the Kubernetes pack YAML. The `imageSwap` block must be under the `pack` section's scope.

 <br />

 ```yaml
 pack:
  imageSwap:
      imageChange: |-
          default:
          # your custom configuration goes here
 ```

>  You can add the `imageSwap` section during the cluster profile creation process or at the cluster deployment. You can customize the image swap functionality in multiple ways. We recommend you review the official [Image Swap configuration](https://github.com/phenixblue/imageswap-webhook/blob/master/README.md#configuration) documentation to learn more. To help you get started, the following are some common configuration patterns.

Reviewer: ```suggestion
 You can add the `imageSwap` section when you create the cluster profile or at cluster deployment. You can customize the image swap functionality several ways. We recommend you review the official [Image Swap configuration](https://github.com/phenixblue/imageswap-webhook/blob/master/README.md#configuration) documentation to learn more. To help you get started, the following are some common configuration patterns.
```
